internal/rbd/encryption.go

```diff
@@ -242,40 +241,46 @@ func (rv *rbdVolume) openEncryptedDevice(ctx context.Context, devicePath string)
 }
 
 func (ri *rbdImage) initKMS(ctx context.Context, volOptions, credentials map[string]string) error {
-	var (
-		err       error
-		ok        bool
-		encrypted string
-	)
+	kmsID, err := ri.ParseEncryptionOpts(ctx, volOptions)
+	if err != nil {
+		return err
+	}
 
-	// if the KMS is of type VaultToken, additional metadata is needed
-	// depending on the tenant, the KMS can be configured with other
-	// options
-	// FIXME: this works only on Kubernetes, how do other CO supply metadata?
-	ri.Owner, ok = volOptions["csi.storage.k8s.io/pvc/namespace"]
-	if !ok {
-		util.DebugLog(ctx, "could not detect owner for %s", ri)
+	err = ri.configureEncryption(kmsID, credentials)
+	if err != nil {
+		return fmt.Errorf("invalid encryption kms configuration: %w", err)
 	}
 
+	return nil
+}
+
+// ParseEncryptionOpts returns kmsID and sets Owner attribute.
+func (ri *rbdImage) ParseEncryptionOpts(ctx context.Context, volOptions map[string]string) (string, error) {
+	var (
+		err              error
+		ok               bool
+		encrypted, kmsID string
+	)
+
 	encrypted, ok = volOptions["encrypted"]
 	if !ok {
-		return nil
+		return "", nil
 	}
-
-	isEncrypted, err := strconv.ParseBool(encrypted)
+	kmsID, err = util.FetchEncryptionKMSID(encrypted, volOptions["encryptionKMSID"])
 	if err != nil {
-		return fmt.Errorf(
-			"invalid value set in 'encrypted': %s (should be \"true\" or \"false\")", encrypted)
-	} else if !isEncrypted {
-		return nil
+		return "", err
 	}
 
-	err = ri.configureEncryption(volOptions["encryptionKMSID"], credentials)
-	if err != nil {
-		return fmt.Errorf("invalid encryption kms configuration: %w", err)
+	// if the KMS is of type VaultToken, additional metadata is needed
+	// depending on the tenant, the KMS can be configured with other
+	// options
+	// FIXME: this works only on Kubernetes, how do other CO supply metadata?
+	ri.Owner, ok = volOptions["csi.storage.k8s.io/pvc/namespace"]
+	if !ok && kmsID != "" {
+		util.DebugLog(ctx, "could not detect owner for %s", ri)
 	}
 
-	return nil
+	return kmsID, nil
```

Comment:
ci tests are failing due to this. 

Should we set rbd.owner for all pvc or only for encrypted pvc @Madhu-1 @nixpanic ? 
(hopefully this refactoring makes more sense and reusable)

Reply:
AFAIK Owner is needed only for the encrypted PVC not for normal PVC. can you be more specific about why this is failing?

Reply:
https://github.com/ceph/ceph-csi/blob/72d56cb8db1feb9f18fe001da480ba1d973a07bf/e2e/rbd.go#L341 
This checks owner on a regular non-encrypted PVC

Reply:
https://github.com/ceph/ceph-csi/pull/2296#discussion_r677145077 

IMO, let's keep previous behavior of setting namespace as owner for all PVCs. 
Debug logs on rbd controller will not matter much (as ceph csi will set owner on all PVC)?